src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -626,17 +631,23 @@ private struct UrlDecoder
             private void FlushBytes()
             {
                 Debug.Assert(_numBytes > 0);
+                if (_charBuffer == null)
+                    _charBuffer = new char[_bufferSize];
+
                 _numChars += _encoding.GetChars(_byteBuffer, 0, _numBytes, _charBuffer, _numChars);
                 _numBytes = 0;
             }
 
             internal UrlDecoder(int bufferSize, Encoding encoding)
             {
+                _containsUnsafe = false;
+                _containsSpaces = false;
+
                 _bufferSize = bufferSize;
                 _encoding = encoding;
 
-                _charBuffer = new char[bufferSize];
-                
+                _charBuffer = null; // char buffer created on demand
```

Comment:
Just a follow up: its due to [this](https://github.com/hughbe/corefx/blob/webutility-decode/src/System.Runtime.Extensions/src/System/Net/WebUtility.cs#L424) line of code.

I don't really understand why "// 7 bit have to go as bytes because of Unicode" but hey